backend/src/apis/round.rs

```diff
@@ -62,58 +58,56 @@ impl MstInclusionProof {
 pub struct Snapshot<const LEVELS: usize, const N_ASSETS: usize, const N_BYTES: usize> {
     mst: MerkleSumTree<N_ASSETS, N_BYTES>,
     assets_state: [Asset; N_ASSETS],
-    trusted_setup: [SetupArtifacts; 2],
+    trusted_setup: SetupArtifacts,
 }
 
-pub struct Round<const LEVELS: usize, const N_ASSETS: usize, const N_BYTES: usize> {
+pub struct Round<'a, const LEVELS: usize, const N_ASSETS: usize, const N_BYTES: usize> {
     timestamp: u64,
     snapshot: Snapshot<LEVELS, N_ASSETS, N_BYTES>,
-    signer: SummaSigner,
+    signer: &'a SummaSigner,
 }
 
 impl<const LEVELS: usize, const N_ASSETS: usize, const N_BYTES: usize>
-    Round<LEVELS, N_ASSETS, N_BYTES>
+    Round<'_, LEVELS, N_ASSETS, N_BYTES>
 where
     [usize; N_ASSETS + 1]: Sized,
     [usize; 2 * (1 + N_ASSETS)]: Sized,
 {
-    pub fn new(
-        signer_key: &str,
-        chain_id: u64,
-        rpc_url: &str,
-        summa_address_input: AddressInput,
-        entry_csv_path: &str,
+    pub fn new<'a>(
```

Comment:
The main point of the PR for me was that `Round` should accept as input an object binded to the `Tree` trait. Similarly to what you applied to `Solvency::init`. But I don't know why you applied it to `Solvency` which is now dismissed

Reply:
Ah, that's what you were looking for. Honestly, I was trying to figure out how to define the `Tree` trait as a concrete type in the `SolvencyCircuit` struct, but I realized it's impossible without using `Box`, which is kind of smart pointer.

Reply:
Why is that?

Can't you define it like this? https://github.com/summa-dev/summa-solvency/blob/enrico-tree-trait/zk_prover/src/circuits/solvency.rs#L76

Reply:
It because the `merkle_sum_tree` in the `solvency::init` evaluates some values, like `root_hash`, not assign into `SolvencyCircuit`. Anyway I fixed like this: 
```Rust
pub struct Snapshot<const LEVELS: usize, const N_ASSETS: usize, const N_BYTES: usize> {
    mst: Box<dyn Tree<N_ASSETS, N_BYTES>>, // Instead of MerkleSumTree<N_ASSETS, N_BYTES>,
    assets_state: [Asset; N_ASSETS],
    trusted_setup: SetupArtifacts,
}
```

Reply:
Updated at [5a9120f](https://github.com/summa-dev/summa-solvency/pull/179/commits/5a9120f25dfebf26826ba5799e4a53c571e8cf06)